### PR TITLE
fix(iroh): improve and test blob share

### LINKED ISSUE #1978: `iroh blob share` is broken
on main as of 2024-01-25 `iroh blob share HASH` returns 
```
Error: failed to retrieve blob info

Caused by:
    Expected header frame
```

### PR DESCRIPTION
**Pull request by dignifiedquire**
- adds `iroh.blobs.share`
- adds `iroh.blobs.status`
 
Closes #1978

### CODE REVIEW

File: iroh/src/client.rs [hunk 1]
```diff
@@ -433,6 +434,51 @@ where
...
+        Ok(BlobStatus {
+            size: reader.size,
+            is_complete: reader.is_complete,
+        })
+    }
+}
+
+/// Status information about a blob.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BlobStatus {
```
Comment (rklaehn) [inline iroh/src/client.rs:477]: Would also prefer an enum here. E.g. "partial" is a pretty wide range from 1/1TiB to 999GiB/1TiB. We might want to add some more info here.

Comment (dignifiedquire) [reply iroh/src/client.rs:477]: but that information is not known, the store doesn't expose this easily

Comment (rklaehn) [reply iroh/src/client.rs:477]: Yes, I know. And it is not free to compute. What I'm saying is that with an enum it is easier to evolve it as more information becomes available.

Comment (dignifiedquire) [reply iroh/src/client.rs]: done

File: iroh/src/client.rs [hunk 2]
```diff
@@ -433,6 +434,51 @@ where
         self.rpc.rpc(BlobDeleteBlobRequest { hash }).await??;
         Ok(())
     }
+
+    /// Share a blob.
+    pub async fn share(
+        &self,
+        hash: Hash,
+        blob_format: BlobFormat,
+        no_derp: bool,
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply iroh/src/client.rs]: done

**Review by rklaehn — APPROVED** [review]

**Pull request merged**